Update required dependencies for Debian/Ubuntu

mfonville opened the pull request:
After trying to build Atom on a very bare Ubuntu install, I figured out some missing dependencies for building: https://github.com/atom/flight-manual.atom.io/pull/683

That also implied these libraries are the actual correct dependencies

mfonville committed:
update required libs

mfonville commented:
@DeeDeeG would you maybe mind checking these for the Red Hat version?

DeeDeeG commented:
@mfonville I know it's likely too late to ask, but it would be great if you could provide any error messages that let you know that these packages were needed. Or the names/paths of the needed files, for instance shared object (`.so`) files. (As I recall we found out in the previous PR, the package names may not be the same in `.rpm` land, but the shared object files have a fair chance of having the same name.)

Or not quite as time-saving, but what minimal environment did you use to find out these packages were needed? (The `debian` or `ubuntu` Docker images, maybe?)

mfonville commented:
@DeeDeeG I unfortunately don't have the exact .so anymore. You can take a look at the PR for the Atom manual I submitted, in addition to that there were only some libraries that I did find, but later seemed to be implied by e.g `gtk3` like: `libxcomposite1` `libxcursor1` `libxdamage1` `libatk1.0-0` `libatk-bridge2.0-0` `libgdk-pixbuf2.0-0`; and `libxtst6` did imply already for `libxi6` 

The testing I did by building Atom on a Ubuntu Server ISO (20.04 in Virtualbox)

mfonville commented:
@DeeDeeG could you check if #22076 works for you on RPM-based distros? Then both PRs can be merged.

DeeDeeG commented:
I haven't really had the time to set aside and take a look at this. I'm afraid I might not for a couple of weeks.

mfonville commented:
Ok, there is no hurry :-)

Imreallycool8 commented:
Is my inbox some sort of message system?

On Tue, Mar 23, 2021 at 11:05 AM Maarten Fonville ***@***.***>
wrote:

> Ok, there is no hurry :-)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/atom/atom/pull/22015#issuecomment-805115037>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATC5MHKY72UTR452VFQLYV3TFDJ6VANCNFSM4Y2JIFBQ>
> .
>

DeeDeeG commented:
@Imreallycool8 perhaps you are subscribed to updates in this repository?

Look for this button at the top of the page:

![GitHub "watch" button](https://user-images.githubusercontent.com/20157115/112201965-394d3700-8be7-11eb-838a-2f6bea17239c.png)

You can click that and select a narrower option, such as "Participating and @mentions" or "Ignore" or "Custom".

Imreallycool8 commented:
Hello, how do I make a new window for just a text box?

mfonville commented:
@darangi You can already safely merge this PR, the other PR for the RPM-dependencies does not have any direct dependency with this one.

Thank you

DeeDeeG commented on resources/linux/debian/control.in:
```diff
@@ -1,6 +1,6 @@
 Package: <%= appFileName %>
 Version: <%= version %>
-Depends: git, libgtk-3-0 (>= 3.9.10), libgcrypt20, libnotify4, libxtst6, libnss3 (>= 2:3.22), libglib2.0-bin | kde-cli-tools | kde-runtime, xdg-utils, libx11-xcb1, libxss1, libxkbfile1, libcurl3 | libcurl4
+Depends: git, libgcrypt20, libgtk-3-0 (>= 3.9.10), libnotify4, libnss3 (>= 2:3.22), libglib2.0-bin | kde-cli-tools | kde-runtime, xdg-utils, libx11-xcb1, libxcb-dri3-0, libxss1, libxtst6, libxkbfile1, libcurl3 | libcurl4
```
Do we actually need `libX11-dri-3-0` or is it `libX11-xcb` like on Fedora/openSUSE?

Maybe `libX11-xcb1`? See: https://packages.ubuntu.com/hirsute/amd64/libx11-xcb1/filelist

DeeDeeG replied:
Okay, I see `libxcb-dri3-0` is actually needed, from some testing in Docker.

DeeDeeG commented:
I'm giving this the old "Docker test". Results later today, if all goes to plan.

DeeDeeG commented:
I can recommend adding `libgbm1` to the requirements. Once that is installed, all tested Debian/Ubuntu versions can do `atom-dev --version --no-sandbox`.

Results:

- Debian
  - 8-slim (End-of-Life): needs `libgbm.so.1` --> `libgbm1` package :x:
  - 9-slim (oldest supported): good, no changes needed :heavy_check_mark:
  - 10-slim (newest supported): needs `libgbm.so.1` --> `libgbm1` package :x:
- Ubuntu
  - 16.04 (recently End-of-Life): good, no changes needed :heavy_check_mark:
  - 18.04 (oldest supported): needs `libgbm.so.1` --> `libgbm1` package :x:
  - 20.04: good, no changes needed :heavy_check_mark:
  - 20.10: good, no changes needed :heavy_check_mark:
  - 21.04 (newest supported): good, no changes needed :heavy_check_mark:

<details><summary>Testing method and notes (click to expand):</summary>

Similar to https://github.com/atom/atom/pull/22076#issuecomment-843338497...

Once the Docker container is up and running, first do `apt-get update && apt-get install nano` to get a working text editor. Use `nano` to write out and save this script, or else `apt-get` will swallow some of the later commands and they won't actually run:

```sh
export DEBIAN_FRONTEND=noninteractive # skip manual configuration prompt during install of the `tzdata` package
apt-get install -y wget unzip
wget 'https://dev.azure.com/DeeDeeG/b/_apis/build/builds/1128/artifacts?artifactName=atom-amd64.deb&api-version=6.0&%24format=zip' -O _atom.zip
unzip _atom.zip
apt-get install -y ./atom-amd64.deb/atom-amd64.deb
atom-dev --version --no-sandbox
/usr/share/atom-dev/atom --version --no-sandbox
apm-dev --version
```

Then run this script with `bash ./[script_filename_here]`.

Note: For the now End-of-Life Debian 8 release, `apt-get` refuses to install a bare `.deb` package. So you have to use `dpkg -i ./atom-amd64.deb/atom-amd64.deb` then `apt-get install -f` to get the downloaded Atom `.deb` file installed, instead of `apt-get install ./atom-amd64.deb/atom-amd64.deb`. Then the rest of the script is the same.

</details>

mfonville committed:
Add libgbm1
promote libasound to requires

mfonville commented:
Ok, I added `libgbm1` too.

Also I changed `libasound2` from recommends to requires, because it is after further checking a strict requirement. Atom cannot start without it.

DeeDeeG reviewed (approved):
Looks good to me. :+1: 

I re-tested Debian 10 and Ubuntu 18.04, which had a missing dependency before, and I also re-tested Ubuntu 20.04 for good measure. All pass the "Docker test" now (test described above). I'm not re-testing the others as they all should pass now, and it takes too long to test all eight.

sadick254 commented:
@DeeDeeG Thank you for the additional tests.

CyberzNet commented:
> After trying to build Atom on a very bare Ubuntu install, I figured out some missing dependencies for building: [atom/flight-manual.atom.io#683](https://github.com/atom/flight-manual.atom.io/pull/683)
> 
> That also implied these libraries are the actual correct dependencies

https://github.com/atom/atom/pull/22015#issue-825034793
